### PR TITLE
Sanlouise 17531 use contactinformationfield directly

### PR DESCRIPTION
**Pull request by sanlouise**
## Description

**The original component structure looked like:**
```
ProfileInfoTable
↓
PhoneField/EmailField/AddressField
↓
ContactInformationField
↓
PhoneEditView/EmailEditView/AddressEditView (or jsx when not editing)
↓
ContactInformationEditView
↓
ContactInfoForm
↓
SchemaForm
```

**After this PR, the structure looks like:**
```
ProfileInfoTable
↓
ContactInformationField
↓
ContactInformationEditView (or ContactInformationView when not editing)
↓
SchemaForm
```

This PR:
- [x] Removes `PhoneField`, `EmailField`, `AddressField` in favor of `ContactInformationField`
- [x] Pulls JSX for modals out of `ContactInformationEditView` and created `CannotEditModal` and `ConfirmCancelModal`
- [x] Removes `ContactInformationEditButton`
- [x] Pulls `formSchema`, `uiSchema`, `title`, `connvertCleanDataToPayload` and `apiRoute` from `deriveContactInfoProperties` so they no longer need to be passed down as props.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/util/deriveContactInfoProperties.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { API_ROUTES, FIELD_TITLES, FIELD_NAMES } from '@@vap-svc/constants';
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/deriveContactInfoProperties.js:1]: Naming things is hard. I like this collection of helpers and such, I'm just not crazy about the name of the file. For example, importing the `phoneNumbers` from 'deriveContactInfoProperties' reads a little weird to me. I don't have a great suggestion for a better name or organization strategy though :(

**Review by erikphansen — COMMENTED** [review]
I like the overall approach of moving some functionality closer to where it's needed and breaking other functionality out into helpers that can be imported where needed. We are starting to get a lot of helpers in the `@@profile/util` folder. Taking a look at maybe organizing all those better might be another thing you could do. Looks good to me! You must have a pretty solid understanding of how the contact info components work by now 😆

File: src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx [hunk 2]
```diff
@@ -173,36 +256,137 @@ class ContactInformationEditView extends Component {
...
+                onCancel={onCancel}
+                onDelete={this.onDelete}
+                title={title}
+                analyticsSectionName={analyticsSectionName}
+                isLoading={isLoading}
+                deleteEnabled={
+                  data && fieldName !== FIELD_NAMES.MAILING_ADDRESS
+                }
+              >
+                <div>
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx:305]: nit: can this be a Fragment instead?

File: src/applications/personalization/profile/components/personal-information/ContactInformationView.jsx [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
   }
 
-  if (type === 'phone') {
+  if (phoneNumbers.includes(fieldName)) {
     return (
       <div>
         <Telephone
-          contact={`${data.areaCode}${data.phoneNumber}`}
-          extension={data.extension}
+          contact={`${data?.areaCode}${data?.phoneNumber}`}
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/ContactInformationView.jsx:28]: no need for checking the existence of `data` here.

File: src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import AddressField from './AddressField';
+import ContactInformationField from '~/applications/personalization/profile/components/personal-information/ContactInformationField';
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx:4]: I know we don't really have guidance for when to use the Babel aliases or not, but I think `from '../ContactInformationField'` would work here?

File: src/applications/personalization/profile/components/personal-information/email-addresses/EmailInformationSection.jsx [hunk 5]
```diff
@@ -3,11 +3,11 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import React from 'react';
 
-import { FIELD_NAMES, FIELD_TITLES } from '@@vap-svc/constants';
+import { FIELD_NAMES } from '@@vap-svc/constants';
+import ContactInformationField from '~/applications/personalization/profile/components/personal-information/ContactInformationField';
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/email-addresses/EmailInformationSection.jsx:7]: Again maybe a relative import would be cleaner here? But that's a matter of opinion.

File: src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx [hunk 6]
```diff
@@ -14,18 +14,13 @@ const AddressesTable = ({ className }) => (
       {
         title: 'Mailing address',
         value: (
-          <AddressField
-            title={FIELD_TITLES[FIELD_NAMES.MAILING_ADDRESS]}
-            fieldName={FIELD_NAMES.MAILING_ADDRESS}
-            deleteDisabled
-          />
+          <ContactInformationField fieldName={FIELD_NAMES.MAILING_ADDRESS} />
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx:17]: I certainly like this :)

File: src/applications/personalization/profile/components/personal-information/phone-numbers/PhoneNumbersTable.jsx [hunk 7]
```diff
@@ -1,49 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import { FIELD_NAMES, FIELD_TITLES } from '@@vap-svc/constants';
-import PhoneField from './PhoneField';
+import { FIELD_NAMES } from '@@vap-svc/constants';
+
 import ProfileInfoTable from '../../ProfileInfoTable';
 
+import ContactInformationField from '~/applications/personalization/profile/components/personal-information/ContactInformationField';
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/personal-information/phone-numbers/PhoneNumbersTable.jsx:8]: another potential relative import here

File: src/applications/personalization/profile/tests/components/ContactInformationField.unit.spec.jsx [hunk 8]
```diff
@@ -59,17 +59,17 @@ describe('<ContactInformationField/>', () => {
     component.unmount();
   });
 
-  // it('renders the ContactInformationEditView', () => {
-  //   props.showEditView = true;
-  //   component = enzyme.shallow(<ContactInformationField {...props} />);
+  it('renders the ContactInformationEditView', () => {
+    props.showEditView = true;
+    component = enzyme.shallow(<ContactInformationField {...props} />);
```
Comment (erikphansen) [inline src/applications/personalization/profile/tests/components/ContactInformationField.unit.spec.jsx:64]: next step: replace all these Enzyme tests with RTL 😆  j/k

File: src/applications/personalization/profile/util/contact-information/formValues.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
+import { ADDRESS_POU, FIELD_NAMES, USA } from '@@vap-svc/constants';
+import ADDRESS_DATA from '~/platform/forms/address/data';
+import {
+  addresses,
+  phoneNumbers,
+} from '~/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes';
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/contact-information/formValues.js:6]: can this just be `from './getContactInfoFieldAttributes'`?

File: src/applications/personalization/profile/util/contact-information/formValues.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import pickBy from 'lodash/pickBy';
+
+/**
+ * Returns a copy of the input object with an added `view:livesOnMilitaryBase`
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/contact-information/formValues.js:10]: deja vu... didn't we fix this incorrect comment in your previous PR?

File: src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
+import { API_ROUTES, FIELD_TITLES, FIELD_NAMES } from '@@vap-svc/constants';
+
+import {
+  emailConvertCleanDataToPayload,
+  emailUiSchema,
+  emailFormSchema,
+} from '~/applications/personalization/profile/util/contact-information/emailUtils';
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js:7]: could be a relative import

File: src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+import {
+  phoneConvertCleanDataToPayload,
+  phoneUiSchema,
+  phoneFormSchema,
+} from '~/applications/personalization/profile/util/contact-information/phoneUtils';
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js:12]: could be a relative import

File: src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+import { addressConvertCleanDataToPayload } from 'applications/personalization/profile/util/contact-information/addressUtils';
```
Comment (erikphansen) [inline src/applications/personalization/profile/util/contact-information/getContactInfoFieldAttributes.js:13]: could be a relative import

**Review by erikphansen — COMMENTED** [review]
Gonna take another pass at this but here are some initial comments!

File: src/applications/personalization/profile/components/personal-information/CannotEditModal.jsx [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        Please go back and save or cancel your work before editing a new section
+        of your profile. If you cancel, your in-progress work won’t be saved.
+      </p>
+      <button onClick={hideCannotEditModal}>OK</button>
+    </Modal>
+  );
+};
+
+CannotEditModal.propTypes = {
+  activeSection: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (sanlouise) [reply src/applications/personalization/profile/components/personal-information/CannotEditModal.jsx:24]: This component is rendered, though hidden when `visible` is `false`, and upon initial render or just when the user has not yet clicked a section to edit, this value is `undefined`.

File: src/applications/personalization/profile/components/personal-information/CannotEditModal.jsx [hunk 15]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+
+const CannotEditModal = props => {
+  const { activeSection, showCannotEditModal, hideCannotEditModal } = props;
```
[the remark this answers is not in the document]
Comment (sanlouise) [reply src/applications/personalization/profile/components/personal-information/CannotEditModal.jsx:6]: I agree, will make the update

File: src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx (hunk 6 above)
Comment (sanlouise) [reply src/applications/personalization/profile/components/personal-information/addresses/AddressesTable.jsx:17]: 🎉

File: src/applications/personalization/profile/util/contact-information/formValues.js (hunk 10 above)
Comment (sanlouise) [reply src/applications/personalization/profile/util/contact-information/formValues.js:10]: We did, thanks for catching that - this must have been its way through my merge conflicts

File: src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx (hunk 2 above)
Comment (sanlouise) [reply src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx:305]: I changed it to a Fragment and it unfortunately messed up the positioning of the buttons, so I reverted to a div.

File: src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx [hunk 16]
```diff
@@ -173,36 +256,137 @@ class ContactInformationEditView extends Component {
...
+              data={field.value}
               uiSchema={field.uiSchema}
-              onUpdateFormData={
-                type === 'address'
-                  ? this.onInput
-                  : this.props.onChangeFormDataAndSchemas
-              }
-              onSubmit={onSubmit}
+              onChange={event => {
+                if (addressFieldNames.includes(fieldName)) {
```
[the remark this answers is not in the document]
Comment (sanlouise) [reply src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx]: Absolutely! Done, much better now :)

**Review by erikphansen — APPROVED** [review]
Feel free to merge. I think we have enough analytics in place to know if something goes horribly wrong 😆

**Pull request merged**